lib/web_ui/lib/src/engine/semantics/scrollable.dart

```diff
@@ -79,6 +99,13 @@ class Scrollable extends RoleManager {
 
   @override
   void update() {
+      // We neutralize the scroll position after all children have been
+      // updated. Otherwise the browser does not yet have the sizes of the
+      // child nodes and resets the scrollTop value back to zero.
+      semanticsObject.owner.addOneTimePostUpdateCallback(() {
+        _neutralizeDomScrollPosition();
```

Comment:
`_neutralizeDomScrollPosition` updates `verticalContainerAdjustment` and `horizontalContainerAdjustment`. Do we need to call `recomputePositionAndSize` after this?

Reply:
I'm not sure. Was `recomputePositionAndSize` meant to be called every time we neutralize the position?

Currently we wait for everything to finish updating before we neutralize the scrolling position. Now is it possible that calling `recomputePositionAndSize` would cause unintended consequences and change the value of ScrollTop/ScrollLeft?

Reply:
Yes. More generally, `recomputePositionAndSize` should be called every time a parameter that affects it changes. Position and size depend on a whole bunch of stuff, like [rect](https://github.com/flutter/engine/blob/93da4784765158a89e269cdea36b2ba7fab85a27/lib/web_ui/lib/src/engine/semantics/semantics.dart#L1307), [transform](https://github.com/flutter/engine/blob/93da4784765158a89e269cdea36b2ba7fab85a27/lib/web_ui/lib/src/engine/semantics/semantics.dart#L1314), [vertical and horizontal adjustments](https://github.com/flutter/engine/blob/93da4784765158a89e269cdea36b2ba7fab85a27/lib/web_ui/lib/src/engine/semantics/semantics.dart#L1320), etc. If any of them change and we don't call `recomputePositionAndSize` we risk the position and size to go out of sync with the current state of the semantics tree.

I see two ways to fix this:

1. Remove `addOneTimePostUpdateCallback`. This should work because [roles are updated](https://github.com/flutter/engine/blob/93da4784765158a89e269cdea36b2ba7fab85a27/lib/web_ui/lib/src/engine/semantics/semantics.dart#L1009) before calling `recomputePositionAndSize`. However, as you mention above, this breaks some tests.
2. Call `recomputePositionAndSize` right after `_neutralizeDomScrollPosition`.

AFAICT, for the purposes of this bug fix either should work. Option 1 is probably a bigger yak shave though.

Reply:
I'll go with option 2 for now.